gpio/imx6ull-gpio/imx6ull-gpio.c

```diff
@@ -255,10 +256,49 @@ void thread(void *arg)
 }
 
 
-int main(void)
+static int parse_args(int argc, char *argv[])
+{
+	int opt;
+	unsigned long prio;
+	char *endptr;
+
+	while ((opt = getopt(argc, argv, "p:")) != -1) {
+		switch (opt) {
+			case 'p':
+				prio = strtoul(optarg, &endptr, 0);
+				if (endptr == optarg) {
```

Comment:
nitpicking, `*endptr == '\0'` means the whole input was valid, Your check won't fail for `-p 3sta`